appeals/web/src/server/lib/mappers/page-components/interested-party-comments/view-and-review/reject.mapper.js

```diff
@@ -0,0 +1,92 @@
+/** @typedef {import("../../../../../appeals/appeal-details/interested-party-comments/interested-party-comments.types.js").RejectionReasons} RejectionReasons */
+
+/**
+ * @param {Appeal} appealDetails
+ * @returns {PageContent}
+ */
+export function rejectInterestedPartyCommentPage(appealDetails) {
+	const shortReference = appealShortReference(appealDetails.appealReference);
+
+	const pageContent = {
+		heading: 'Why are you rejecting the comment?',
+		backLinkUrl: `/appeals-service/appeal-details/${appealDetails.appealId}/review`,
+		preHeading: `Appeal ${shortReference}`,
+		hint: 'Select all that apply.',
+		headingClasses: 'govuk-heading-l'
+	};
+
+	return pageContent;
+}
+
+/**
+ * @param {Representation} comment - The comment containing the selected rejection reasons.
+ * @param {RepresentationRejectionReason[]} rejectionReasonOptions - The available rejection reason options.
+ * @returns {import('../../../../../appeals/appeals.types.js').CheckboxItemParameter[]}
+ */
+export function mapRejectionReasonOptionsToCheckboxItemParameters(comment, rejectionReasonOptions) {
+	const existingReasonIds = comment.rejectionReasons.map((reason) => reason.id);
+
+	return rejectionReasonOptions.map((reason) => {
+		const isChecked = existingReasonIds.includes(reason.id);
+		const selectedReason = comment.rejectionReasons.find((r) => r.id === reason.id);
+		const existingText = selectedReason?.text || [''];
+
+		const item = {
+			value: reason.id.toString(),
+			text: reason.name,
+			checked: isChecked,
+			hasText: reason.hasText,
+			textItems: existingText
+		};
+
+		return item;
+	});
+}
+
+/**
+ * @param {RejectionReasons} rejectionReasons
+ * @returns {RejectionReasonUpdateInput[]}
+ */
+export function mapRejectionReasonPayload(rejectionReasons) {
+	/** @type {Record<number, RejectionReasonUpdateInput>} */
+	const mappedReasons = {};
+
+	if (typeof rejectionReasons.rejectionReason === 'string') {
+		const id = parseInt(rejectionReasons.rejectionReason, 10);
+		if (!isNaN(id)) {
+			mappedReasons[id] = { id, text: [] };
+		}
+	} else if (Array.isArray(rejectionReasons.rejectionReason)) {
+		rejectionReasons.rejectionReason.forEach((id) => {
+			const numId = parseInt(id, 10);
+			if (!isNaN(numId)) {
+				mappedReasons[numId] = { id: numId, text: [] };
+			}
+		});
+	}
+
+	Object.entries(rejectionReasons).forEach(([key, value]) => {
+		const reasonIdMatch = key.match(/^rejectionReason-(\d+)$/);
+		if (reasonIdMatch) {
+			const id = parseInt(reasonIdMatch[1], 10);
+			if (!isNaN(id)) {
+				if (!mappedReasons[id]) {
+					mappedReasons[id] = { id, text: [] };
+				}
+				if (Array.isArray(value)) {
+					mappedReasons[id].text = value.filter((text) => text && text.trim() !== '');
+				} else if (typeof value === 'string' && value.trim() !== '') {
+					mappedReasons[id].text = [value];
+				}
+			}
+		}
+	});
+
+	return Object.values(mappedReasons);
+}
```

Comment:
I'm not sure I understand exactly what this does, need to look at it more in the morning, but
```suggestion
/**
 * @template T
 * @param {T | T[]} maybeArray
 * @return {T[]}
 */
const enforceArray = (maybeArray) => (Array.isArray(maybeArray) ? maybeArray : [maybeArray]);

/**
 * @param {RejectionReasons} rejectionReasons
 * @returns {RejectionReasonUpdateInput[]}
 */
export function mapRejectionReasonPayload2({ rejectionReason: rawRejectionReasonIds, ...reasons }) {
	if (!rawRejectionReasonIds) return [];

	/** @type {Record<number, RejectionReasonUpdateInput>} */
	const mappedReasons = {};

	const rejectedReasonIds = enforceArray(rawRejectionReasonIds);

	rejectedReasonIds.forEach((id) => {
		const numId = parseInt(id, 10);
		if (isNaN(numId)) return;
		// TODO remove this mutation
		mappedReasons[numId] = { id: numId, text: [] };
	});

	Object.entries(reasons).forEach(([key, rawValue]) => {
		if (!rawValue) return;

		const [, reasonIdMatch] = /^rejectionReason-(\d+)$/.exec(key) || [];
		if (!reasonIdMatch) return;

		const id = parseInt(reasonIdMatch, 10);
		if (isNaN(id)) return;

		const value = enforceArray(rawValue);
		const text = value.filter((text) => text && text.trim() !== '');
		mappedReasons[id] = { id, text };
	});

	return Object.values(mappedReasons);
}
```

here it is in 6 less lines with much less nesting

```suggestion
/**
 * @param {RejectionReasons} rejectionReasons
 * @returns {RejectionReasonUpdateInput[]}
 */
export function mapRejectionReasonPayload3({ rejectionReason: rawRejectionReasonIds, ...reasons }) {
	if (!rawRejectionReasonIds) return [];

	const rejectedReasonIds = enforceArray(rawRejectionReasonIds);

	/** @type {Record<number, RejectionReasonUpdateInput>} */
	const firstMappedReasons = rejectedReasonIds.reduce((acc, rawId) => {
		const id = parseInt(rawId, 10);
		if (isNaN(id)) return acc;
		return { ...acc, [id]: { id, text: [] } };
	}, {});

	/** @type {Record<number, RejectionReasonUpdateInput>} */
	const secondMappedReasons = Object.entries(reasons).reduce((acc, [key, rawValue]) => {
		if (!rawValue) return acc;

		const [, reasonIdMatch] = /^rejectionReason-(\d+)$/.exec(key) || [];
		if (!reasonIdMatch) return acc;

		const id = parseInt(reasonIdMatch, 10);
		if (isNaN(id)) return acc;

		const value = enforceArray(rawValue);
		const text = value.filter((text) => text && text.trim() !== '');
		return { ...acc, [id]: { id, text } };
	}, firstMappedReasons);

	return Object.values(secondMappedReasons);
}
```
and here it is in 28 lines without mutation. I'm not totally convinced we even need `firstMappedReasons` so maybe we can cut it right down but I need to test it